src/state/hooks.ts

```diff
@@ -55,6 +56,37 @@ export const usePollBlockNumber = () => {
 
 // Farms
 
+// Fetches farms' data once
+export const useGetFarmsData = () => {
+  const dispatch = useAppDispatch()
+
+  useEffect(() => {
+    dispatch(fetchFarmsData())
+  }, [dispatch])
+}
+
+export const useFastFarmPolling = (pid: number) => {
+  const { fastRefresh } = useRefresh()
+  const dispatch = useAppDispatch()
+
+  useEffect(() => {
+    if (pid) {
+      dispatch(fetchFarmData(pid))
+    }
+  }, [pid, fastRefresh, dispatch])
+}
+
+export const useSlowFarmPolling = (pid: number) => {
+  const { slowRefresh } = useRefresh()
+  const dispatch = useAppDispatch()
+
+  useEffect(() => {
+    if (pid) {
+      dispatch(fetchFarmData(pid))
+    }
+  }, [pid, slowRefresh, dispatch])
+}
```

Comment:
In the future I want to modify `useRefresh` to allow custom values. For now will compose these hooks.